### ISSUE TITLE
Fail early if siteConfig not found (and for other errors)

### ISSUE THREAD

**Issue by jhpedemonte** [maqetta/maqetta#3705]
Currently, if **maqetta.siteConfigDirectory** isn't set correctly, or is set to a dir that doesn't exist or to which the process doesn't have access, a user won't notice that until he/she logs in and Maqetta hangs.  If they know where to look, they may see the stack showing an error in **GetInitializationInfo.java** -- but since that doesn't really mention **siteConfig**, they won't know what to do.

For this property in particular, and maybe for others as well, the server should fail quickly and with details information.  For `siteConfigDirectory`, that probably means that we have to check for that property on startup and try to create a `new File()` with it.  If it fails, explain to the user what they need to do.

Are there any other properties that require such treatment?

@billreed63

**Comment by jhpedemonte** [comment]
-> @peller

**Comment by peller** [comment]
I think the problem is that we do fail early, but we fail silently.  dup of #3691

**Comment by billreed63** [comment]
These are site configuration issues, that can not be addressed by the average user, I would suggest that we need administration pages that the site administrator would use to check the health and proper configuration of the server. 
The user could login to this page and this page/code would integrate the site configuration.
@JonFerraiolo

**Comment by jhpedemonte** [comment]
I don't think this is a dup of #3691 -- I'm talking about failing on the server side, so the person who is setting up the server knows that something is wrong.

I agree with Bill that surfacing this error on the client side isn't the right thing to do.  However, we should have some general "Server is misconfigured.  Please contact the admin." error on the client, rather than silently hanging.

**Comment by peller** [comment]
Agreed.

**Comment by peller** [comment]
We should now have decent error handling, an exception with a descriptive message, and some minimal error UI on the client so it's clear something is wrong.

The remaining issue is whether / how we force the error during install rather than waiting for the first Maqetta launch

-> M10

**Comment by peller** [comment]
-> M11 @jhpedemonte   May want to lower priority
